Emby.Server.Implementations/Plugins/PluginManager.cs

```diff
@@ -0,0 +1,665 @@
+                        {
+                            Id = instance.Id,
+                            Status = PluginStatus.Active,
+                            Name = instance.Name,
+                            Version = instance.Version.ToString()
+                        })
+                    {
+                        Instance = instance
+                    };
+
+                    _plugins.Add(plugin);
+
+                    plugin.Manifest.Status = PluginStatus.Active;
+                }
+                else
+                {
+                    plugin.Instance = instance;
+                    var manifest = plugin.Manifest;
+                    var pluginStr = plugin.Instance.Version.ToString();
+                    bool changed = false;
+                    if (string.Equals(manifest.Version, pluginStr, StringComparison.Ordinal))
+                    {
+                        // If a plugin without a manifest failed to load due to an external issue (eg config),
+                        // this updates the manifest to the actual plugin values.
+                        manifest.Version = pluginStr;
+                        manifest.Name = plugin.Instance.Name;
+                        manifest.Description = plugin.Instance.Description;
+                        changed = true;
+                    }
+
+                    changed = changed || manifest.Status != PluginStatus.Active;
+                    manifest.Status = PluginStatus.Active;
+
+                    if (changed)
+                    {
+                        SaveManifest(manifest, plugin.Path);
+                    }
+                }
+
+                _logger.LogInformation("Loaded plugin: {PluginName} {PluginVersion}", plugin.Name, plugin.Version);
+
+                return instance;
+            }
+#pragma warning disable CA1031 // Do not catch general exception types
+            catch (Exception ex)
+#pragma warning restore CA1031 // Do not catch general exception types
+            {
+                _logger.LogError(ex, "Error creating {Type}", type.FullName);
+                if (plugin != null)
+                {
+                    if (ChangePluginState(plugin, PluginStatus.Malfunctioned))
+                    {
+                        _logger.LogInformation("Plugin {Path} has been disabled.", plugin.Path);
+                        return null;
+                    }
+                }
+
+                _logger.LogDebug("Unable to auto-disable.");
+                return null;
+            }
+        }
+
+        private void UpdatePluginSuperceedStatus(LocalPlugin plugin)
+        {
+            if (plugin.Manifest.Status != PluginStatus.Superceded)
+            {
+                return;
+            }
+
+            var predecessor = _plugins.OrderByDescending(p => p.Version)
+                .FirstOrDefault(p => p.Id.Equals(plugin.Id) && p.IsEnabledAndSupported && p.Version != plugin.Version);
+            if (predecessor != null)
```

Comment:
At startup - it checks that each plugin marked as superceded is still in that state.

It's partly there in case of users manually deleting a folder.